javaparser-core/src/main/java/com/github/javaparser/JavaParser.java

```diff
@@ -100,6 +101,23 @@ private GeneratedJavaParser getParserForProvider(Provider provider) {
         return astParser;
     }
 
+    private String providerToString(Provider provider) {
+        // FIXME: Surely there's a better way...?
```

Comment:
Commenting this line to flag this up for attention -- this method consumes the whole provider stream and returns a string.

Reply:
I've done removed this and replaced it with a preprocessor - marking as "resolved".